### PR TITLE
Initial Minting logic

### PR DESCRIPTION
**Pull request by calebhskim**
Initial minting logic. Adds in a minting button in the nav menu along with the corresponding minting screen. It's a page that allows users to mint tokens. There are two main buttons: "mint" and "connect wallet". Error handling is done through notifications in the right hand corner. Here are some screenshots. Let me know if you guys want more context.

Screen Shot 2021-08-30 at 10 01 37 PM
Screen Shot 2021-08-30 at 10 01 46 PM
Screen Shot 2021-08-30 at 10 02 05 PM
Screen Shot 2021-08-30 at 10 02 36 PM

### CODE REVIEW

File: src/Mint.js [hunk 1]
```diff
@@ -0,0 +1,354 @@
+import { Fragment, useState } from 'react';
+import Web3 from "web3";
+import { ExclamationCircleIcon, XIcon } from '@heroicons/react/solid';
+import { Transition } from '@headlessui/react';
+
+import gif from './assets/148-167_40_04.gif';
+import TheFacesFund from './abi/TheFacesFund.json';
+
+const defaultErrorTitle = 'Whoops!';
```
Comment (shivaalroy) [inline src/Mint.js]: nit: "whoops!" sounds a little aloof, maybe just omit this and say "Looks like something went wrong!" ?

File: src/Mint.js [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+						setWalletAddress(wallet);
+						setSignedIn(true);
+						getContractData(wallet);
+					}
+				}
+				catch (e) {
+					// TODO: Handle error.
+					if (e.code === 4001) {
+						// User rejected request
+						setErrorMessage('It looks like your wallet connection request failed! If you would like to mint a token please try again.');
```
Comment (shivaalroy) [inline src/Mint.js]: maybe say something that indicates they need to accept the request in their wallet extension?

File: src/Mint.js [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+				setShowNote(true);
+			}
+		}
+		else {
+			setErrorMessage('It looks like you may not have Metamask installed. Please see the instructions below to connect your wallet.');
+			setShowNote(true);
+		}
+	}
+
+	// Do we need a sign out function?
```
Comment (shivaalroy) [inline src/Mint.js]: I think a user can just disconnect their MetaMask from our website if they want to sign out

File: src/Mint.js [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+										)}
+										<div className="sm:flex">
+											{saleStarted && <div className="min-w-0 flex-1">
+												<label htmlFor="text" className="sr-only">
+													Number of tokens to mint
+												</label>
+												<div className="relative rounded-md">
+													<input
+														id="num-tokens"
+														type="text"
```
Comment (shivaalroy) [inline src/Mint.js]: should this be type number?

File: src/Mint.js [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+														placeholder="Number of tokens to mint"
+														className="block w-full px-4 py-3 rounded-md border-0 text-base text-gray-900 placeholder-gray-500 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-300 focus:ring-offset-gray-900"
+														min="0"
+														max="20"
+														value={numTokensToPurchase}
+														onChange={ e => {
+															setNumTokensToPurchase(e.target.value || '');
+															setValidNumTokens(e.target.value > 0 && e.target.value < 21);
```
Comment (shivaalroy) [inline src/Mint.js:233]: not sure what this function is doing, but if the page default is 1, and they want to delete the 1 and type 20, will they be able to do so? some website have "validation" that doesn't allow basic typing

File: src/Mint.js [hunk 6]
```diff
@@ -0,0 +1,354 @@
...
+								<rect x={0} y={0} width={4} height={4} className="text-gray-200" fill="currentColor" />
+							</pattern>
+							</defs>
+							<rect width={404} height={384} fill="url(#7a00fe67-0343-4a3c-8e81-c145097a3ce0)" />
+						</svg>
+						<div className="relative md:bg-white md:p-6">
+							<div className="lg:grid lg:grid-cols-2 lg:gap-6">
+								<div className="prose prose-indigo prose-lg text-gray-500 lg:max-w-none">
+									<p>
+										The Faces Fund is a set of 3000 NFTs (Non-Fungible Tokens) focused on community driven charity.
```
Comment (shivaalroy) [inline src/Mint.js]: nit: 3,000

File: src/Mint.js (hunk 1 above)
Comment (calebhskim) [reply src/Mint.js]: Yeah that makes sense. It also shortens the other messages

File: src/Mint.js (hunk 3 above)
Comment (calebhskim) [reply src/Mint.js]: Yeah after some research this isn't really a thing. There are ways to do it but for now i'll ignore it

File: src/Mint.js (hunk 4 above)
Comment (calebhskim) [reply src/Mint.js]: Good catch! Setting this type to `number` adds extra validation to only allow number

File: src/Mint.js (hunk 5 above)
Comment (calebhskim) [reply src/Mint.js:233]: @shivaalroy yes they can type in any number but if it's invalid then they will see a little error message. We can change the behavior if we think it's too noisy. What do you mean by not allow basic typing?

File: src/Mint.js (hunk 6 above)
Comment (calebhskim) [reply src/Mint.js]: Thanks!

**Pull request merged**